spidermon/contrib/scrapy/monitors.py

```diff
@@ -0,0 +1,147 @@
+from spidermon import monitors, MonitorSuite, Monitor
+from spidermon.exceptions import NotConfigured
+from ..monitors.mixins.spider import SpiderMonitorMixin
+
+SPIDERMON_MIN_ITEMS = 'SPIDERMON_MIN_ITEMS'
+SPIDERMON_MAX_ERRORS = 'SPIDERMON_MAX_ERRORS'
+SPIDERMON_EXPECTED_FINISH_REASONS = 'SPIDERMON_EXPECTED_FINISH_REASONS'
+SPIDERMON_UNWANTED_HTTP_CODES = 'SPIDERMON_UNWANTED_HTTP_CODES'
+
+
+class BaseScrapyMonitor(Monitor, SpiderMonitorMixin):
+    longMessage = False
+
+    def get_settings(self, name, default=None):
+        """Return a settings prioritized by:
+
+            #. ``{name}`` in the ``crawler.settings`` object
+            #. If ``crawler.settings.{name}`` is a dict, try to get the
+               ``{spider.name}`` key of that dictionary, this allows you to
+               have a different configuration per spider in a settings level,
+               like:
+
+               .. code-block:: python
+
+                   SPIDERMON_MAX_ERRORS = {
+                        'my_spider': 100,
+                        'other_spider': 10,
+                   }
+
+        """
+        settings_value = self.crawler.settings.get(name.upper())
+        if type(settings_value) == dict:
+            settings_value = settings_value.get(self.spider.name, None)
+        return settings_value or default
+
+
+@monitors.name('Extracted Items Monitor')
+class ItemCountMonitor(BaseScrapyMonitor):
+    """Check if spider extracted the minimum number of items.
+    You can configure it using ``SPIDERMON_MIN_ITEMS`` setting.
+    There's **NO** default value for this setting, if you try to use this
+    monitor without setting it, it'll raise a ``NotConfigured`` exception.
+    """
+    def run(self, result):
```

Comment:
@raphapassini, any specific reason to use `run` instead of `__init__`?
Also, `run` is executed when the spider is started or at the end of the job?

Reply:
When the ``__init__`` is called you don't have the ``data`` information where you can get the ``crawler``, ``settings``, ``spider`` etc. It __doesn't__ work in the ``__init__`` method.